OpenRA.Game/GameRules/WeaponInfo.cs

```diff
@@ -177,17 +198,33 @@ public bool IsValidAgainst(FrozenActor victim, Actor firedBy)
 		}
 
 		/// <summary>Applies all the weapon's warheads to the target.</summary>
-		public void Impact(Target target, Actor firedBy, IEnumerable<int> damageModifiers)
+		public void Impact(Target impactTarget, WarheadArgs args)
```

Comment:
Why rename `target` to `impactTarget`? IMO the `impact` is already implicit from the method name.